content/python/concepts/matplotlib/terms/legend/legend.md

```diff
@@ -0,0 +1,90 @@
+---
+
+The **`.legend()`** function returns a legend on the axes of a plot or subplot.
+
+## Syntax
+
+```pseudo
+plt.legend(handles, labels)
+```
+
+`legend()` can be called without any parameters. This will simply attempt to add the elements automatically. Both `handles` and `labels` are optional. `handles` is utilized if more control is desired with each entry on the plot and how they are labeled. `labels` is a list of string(s) that will display the labels in the order that they are entered with no regard to the actual entries on the plot.
+
+`.legend()` has the following parameters:
+
+|  Parameter Name  | Data Type(s)         | Usage                                                                                                  |
+| :--------------: | -------------------- | ------------------------------------------------------------------------------------------------------ |
+|    `handles`     | sequence of `Artist` | `handles` is utilized if more control is desired with each entry on the plot and how they are labeled. |
+|     `labels`     | list of string       | `labels` is a list of strings that will display the labels in the order they are entered.              |
+|      `loc`       | int, string          | `loc` is used to determine the position of the legend on the axes. Defaults to `best`.                 |
+|      `ncol`      | int                  | `ncol` is used to determine how many columns the legend will have. Defaults to 1.                      |
+|    `fontsize`    | int, string          | `fontsize` is used change the font size the labels. Defaults to 11.                                    |
+|   `labelcolor`   | list, string, `None` | `labelcolor` is used to change the color of the label(s). Defaults to `None`                           |
+|     `title`      | string, `None`       | `title` is used to insert a title for the legend. Defaults to `None`.                                  |
+| `title_fontsize` | int, string, `None`  | `title_fontsize` is used to change the font size of the legend `title`. Defaults to `None`.            |
+
+## Example
+
+In the following example, the `.legend()` method will be shown using the iris data set which is available using seaborn or pandas. In the first example the `handles` parameter will be demonstrated along with using seaborn to import the dataset. In the second example `legend()` will be demonstrated without using the `handles` parameter along with using pandas to import the dataset:
+
+### Example 1
+
+---
+
+```py
+import matplotlib.pyplot as plt
+import seaborn as sns
+
+iris = sns.load_dataset('iris')
+
+fig, ax = plt.subplots(figsize=(12,6))
+
+sepal_lw = ax.scatter(iris['sepal_length'], iris['sepal_width'], label='Sepal')
+petal_lw = ax.scatter(iris['petal_length'], iris['petal_width'], label='Petal')
+
+plt.legend(handles=[sepal_lw, petal_lw]);
+```
+
+The output from this instance of the `.legend()` method is shown below:
+
+![Example1](/media/matplotlib-legend-1.png)
```

Comment:
I'd go with the example that is considered a better practice. If either modules are a good practice to use, and you wish to keep both examples, I'd just go into what the difference is. Maybe one is better for a scenario.

Reply:
Alternatively, instead of including pandas here, you can cite the existing entries:

- https://www.codecademy.com/resources/docs/pandas/dataframe
OR
- https://www.codecademy.com/resources/docs/pandas/built-in-functions/read-csv